ci/tests/api-compatibility-test.sh

```
@@ -0,0 +1,333 @@
+    
+    # Check if CRD exists
+    if ! oc get crd "$crd_name" &>/dev/null; then
+        log FAIL "CRD not found for $kind: $crd_name"
+        return 1
+    fi
+    
+    # Get supported versions
+    local supported_versions=$(oc get crd "$crd_name" -o jsonpath='{.spec.versions[*].name}')
+    
+    # Check if expected version is supported
+    if [[ " $supported_versions " =~ " ${expected_version##*/} " ]]; then
```

Shellcheck (0.10.0)

[warning] Remove quotes from right-hand side of =~ to match as a regex rather than literally.

(SC2076)